File: python/605. Can Place Flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """
        i = 0 
        plantable = 0
        while (i<len(flowerbed)):
            checkright = False
            checkleft = False

            if (flowerbed[i] != 1):
                 
                if ( i == 0):
                    checkleft=True
                else:
                    if(flowerbed[i-1] == 0):
                        checkleft=True
                if (i == len(flowerbed)-1):
                    checkright = True
                else:
                    if(flowerbed[i+1] ==0):
                         checkright = True

                if ( checkleft and checkright):
                    plantable +=1
                    flowerbed[i] = 1
            i += 1
        return (n <=plantable)    
        

File: javascript/605. Can Place Flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function (flowerbed, n) {

    let i = 0;
    while (i < flowerbed.length) {
        if (flowerbed[i] == 0) {
            let leftcheck = false;
            let rightcheck = false;

            if (i == 0)
                leftcheck = true;
            else {
                if (flowerbed[i - 1] == 0)
                    leftcheck = true
            }
            if (i == flowerbed.length - 1)
                rightcheck = true;
            else {
                if (flowerbed[i + 1] == 0)
                    rightcheck = true
            }
            if (rightcheck && leftcheck) {
                flowerbed[i] = 1;
                n = n - 1;
            }

        }
        i += 1;
    }
   
        return n <= 0;

};